File: Solutions/Frequency-of-the-Most-Frequent-Element.py
from typing import List

class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()
        total = 0
        left = 0
        ans = 1
        
        for i in range(1, len(nums)):
            total += (nums[i] - nums[i - 1]) * (i - left)
            while total > k:
                total -= nums[i] - nums[left]
                left += 1
            ans = max(ans, i - left + 1)
                
        return ans

S = Solution()
print(S.maxFrequency([3,6,9],2))

File: Solutions/Frequency-of-the-Most-Frequent-Element.cpp
class Solution
{
  public:
    int maxFrequency(vector<int> &nums, int k)
    {
        sort(nums.begin(), nums.end());
        int64_t left = 0, right = 0, ans = 1, n = nums.size();
        while (++right < n)
        {
            k -= (right - left) * (nums[right] - nums[right - 1]);
            while (k < 0)
                k += (int64_t)(nums[right] - nums[left++]);
            ans = max(ans, right - left + 1);
        };
        return ans;
    }
};